modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -604,4 +608,200 @@ private void deleteApplication(String applicationId) throws Exception {
 
         oAuth2RestClient.deleteApplication(applicationId);
     }
+
+    @Test(dependsOnMethods = "testDeleteOrganization")
+    public void createOrganizationsForPaginationTests() throws JSONException {
+
+        int createdOrganizationsCount = createOrganizations(totalCreatedOrganizations);
+
+        if (createdOrganizationsCount != totalCreatedOrganizations) {
+            throw new RuntimeException("Failed to create the expected number of organizations for testing.");
+        }
+    }
+
+    @DataProvider(name = "paginationLimitsDataProvider")
+    public Object[][] paginationLimitsDataProvider() {
+
+        return new Object[][]{
+                {10},
+                {20},
+                {25},
+        };
+    }
+
+    @Test(dataProvider = "paginationLimitsDataProvider", dependsOnMethods = "createOrganizationsForPaginationTests")
+    public void testGetPaginatedOrganizationsWithLimit(int limit) {
+
+        String endpointURL = ORGANIZATION_MANAGEMENT_API_BASE_PATH + LIMIT_ARG + limit;
+        Response response = getResponseOfGetWithOAuth2(endpointURL, m2mToken);
+
+        validateHttpStatusCode(response, HttpStatus.SC_OK);
+
+        int actualOrganizationCount = response.jsonPath().getList("organizations").size();
+        int expectedOrganizationCount = Math.min(limit, totalCreatedOrganizations);
+        Assert.assertEquals(actualOrganizationCount, expectedOrganizationCount);
+
+        String nextLink = response.jsonPath().getString("links.find { it.rel == 'next' }.href");
```

Comment:
Can we remove the logic written in the string and extract it outside?

Reply:
Done